website/src/components/Sidebar.js

```diff
@@ -1,10 +1,23 @@
-import React from 'react'
+import React, { useState } from 'react'
+import SearchBar from './SearchBar';
 
 function Sidebar ({children = []}) {
+    const [searchTerm,setSearchTerm] = useState(''); 
+    // console.log(children)
+    const filtered = 
+    children.filter(val => {
```

Comment:
This line throws an error when the website is being compiled. The reason is that you return a value inside the `if` and `else if` conditions, however, if an item does not comply with either of the issues nothing is returned.

So, at the end of the callback of `.filter` you should add a default return value.

Reply:
Also, this entire block `filtered = children.filtered` should be added inside `useEffect` which depends on `searchTerm`:

```js
useEffect(() => {
  //code here
}, [searchTerm]);
```

And I think `filtered` should be a state (and maybe renamed to searchResults):

```
const [searchResults, setSearchResults] = useState(children);
```

Then, inside `useEffect` that you will add, instead of:

```
const filtered = children.filter
```

You will use `setSearchResults` to set the new search results

Reply:
> Also, this entire block `filtered = children.filtered` should be added inside `useEffect` which depends on `searchTerm`:
> 
> ```js
> useEffect(() => {
>   //code here
> }, [searchTerm]);
> ```
> 
> And I think `filtered` should be a state (and maybe renamed to searchResults):
> 
> ```
> const [searchResults, setSearchResults] = useState(children);
> ```
> 
> Then, inside `useEffect` that you will add, instead of:
> 
> ```
> const filtered = children.filter
> ```
> 
> You will use `setSearchResults` to set the new search results

I'll just learn useEffect and try to complete this in a day or two